includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -94,22 +94,25 @@ public function __construct() {
 			return;
 		}
 
-		add_action( 'template_redirect', [ $this, 'set_session' ] );
 		$this->init();
 	}
 
 	/**
 	 * Checks whether authentication is required for checkout.
 	 *
-	 * @since 5.1.0
+	 * @since   5.1.0
+	 * @version x.x.x
 	 *
 	 * @return bool
 	 */
 	public function is_authentication_required() {
-		// If guest checkout is disabled, authentication might be required.
-		if ( 'no' === get_option( 'woocommerce_enable_guest_checkout', 'yes' ) ) {
-			// If account creation is not possible, authentication is required.
-			return ! $this->is_account_creation_possible();
+		// If guest checkout is disabled and account creation is not possible, authentication is required.
+		if ( 'no' === get_option( 'woocommerce_enable_guest_checkout', 'yes' ) && ! $this->is_account_creation_possible() ) {
+			return true;
+		}
+		// If cart contains subscription and account creation is not posible, authentication is required.
+		if ( $this->has_subscription_product() && ! $this->is_account_creation_possible() ) {
+			return true;
```

Comment:
> and no way to create an account is offered.

Either guest checkout must be enabled, or there must be a way to create an account for both the checkout and the PRB to work. I guess you have interpreted the comments as creating an account in general, but in the comment I meant "creating an account upon checkout". I've added that to the comment for clarity in 6ae5a284551eb61f89936a3e735b14820f45eca3

So yes, we're offering a dialog and redirecting merchants to either login or create an account themselves (from my account), but by design, account creation or guest checkout must be enabled in the website, otherwise no one can purchase 🙂 . We're basically copying WC core behavior.